src/ripple/app/misc/Validations.cpp

```diff
@@ -429,25 +429,34 @@ class ValidationsImp : public Validations
         bool anyNew = false;
 
         JLOG (j_.info()) << "Flushing validations";
-        ScopedLockType sl (mLock);
-        for (auto& it: mCurrentValidations)
         {
-            if (it.second)
-                mStaleValidations.push_back (it.second);
-
-            anyNew = true;
-        }
-        mCurrentValidations.clear ();
+            ScopedLockType sl (mLock);
+            for (auto& it: mCurrentValidations)
+            {
+                if (it.second)
+                {
+                    mStaleValidations.push_back (it.second);
+                    anyNew = true;
+                }
+            }
+            mCurrentValidations.clear ();
 
-        if (anyNew)
-            condWrite ();
+            // If there isn't a write in progress already, then write to the
+            // database synchronously.
+            if (anyNew && !mWriting)
```

Comment:
I will go ahead and pass that lock to `condWrite()`.  This stuff is hard enough to think about that we shouldn't make it unnecessarily obscure.